crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -49,10 +49,37 @@ impl BlobVec {
         drop: Option<unsafe fn(OwningPtr<'_>)>,
         capacity: usize,
     ) -> BlobVec {
+        /*NOTE: dangling pointers still need to be well aligned for the type when using slices (even though they are 0-length).
+                This is important for [`SimdAlignedVec`] and any function that would return a slice view of this BlobVec.
+                It is also important if the BlobVec contains ZSTs and is mutably queried, as this will entail the
+                creation of a slice (even if 0-length).
+
+                Since neither strict_provenance nor alloc_layout_extra is stable, there is no way to construct a NonNull::dangling()
+                pointer from `item_layout`.  Creating an aligned NonNull dangling pointer manually with integer casts works, but Miri doesn't like that,
+                and it could break in the future.  Instead, assume that `MAX_SIMD_ALIGNMENT` will be the greatest alignment required of any
+                possible instruction on the system and create a dangling pointer out of a type guaranteed to need that alignment.
+
+                Currently, that type is batch::AlignedBatch64.
+
+                TODO: In the future, this restriction can be easily removed when either `strict_provenance` or `Layout::dangling()` becomes stable.
+
+
+                Obviously, this won't make sense if the contained type needs more alignment than that.
+                Therefore we have the following check:
+        */
+
+        if item_layout.align() > batch::MAX_SIMD_ALIGNMENT {
+            panic!(
+                "Attempted to create a BlobVec with an item layout alignment > MAX_SIMD_ALIGNMENT"
+            );
+        }
```

Comment:
Thanks for the feedback -- I had code like this originally and Miri was throwing a warning, but since you're cool with `-Zmiri-permissive-provenance` I'll go ahead and make that change.

Reply:
CI already runs with `-Zmiri-permissive-provenance` ideally we'd use `ptr::invalid(item_layout.align().max(64))` but thats unstable so oh well :sweat_smile:

Reply:
Done, thank you!